paimon-core/src/main/java/org/apache/paimon/disk/ExternalBuffer.java

```diff
@@ -147,6 +163,7 @@ private void spill() throws IOException {
                                 ? inMemoryBuffer.getNumBytesInLastBuffer()
                                 : segment.size();
                 writer.writeBlock(Buffer.create(segment, bufferSize));
+                numEstimatedBytes += bufferSize;
             }
             LOG.info(
```

Comment:
You can just cache `writer.getSize()`.

Reply:
Get it, directly use `FileChannel.getSize()` can avoid an extra system call to get the file length, Right ?